src/metabase/driver/druid/query_processor.clj

```diff
@@ -746,19 +758,30 @@
          (map (comp update-ts-fn :event))
          (post-process-map projections))))
 
-(defmethod post-process ::total   [_ projections timezone results]
+(defmethod post-process ::total   [_ projections _ results]
   (post-process-map projections (map :result results)))
 
-(defmethod post-process ::topN    [_ projections timezone results]
-  (post-process-map projections (-> results first :result)))
-
-(defmethod post-process ::groupBy [_ projections timezone results]
-  (post-process-map projections (map :event results)))
-
-(defmethod post-process ::timeseries [_ projections timezone results]
+(defmethod post-process ::topN    [_ projections {:keys [middleware]} results]
+  (post-process-map projections
+                    (let [results (-> results first :result)]
+                      (if (false? (:format-rows? middleware))
+                        (map #(u/update-when % :timestamp parse-timestamp) results)
+                        results))))
+
+(defmethod post-process ::groupBy [_ projections {:keys [middleware]} results]
+  (post-process-map projections
+                    (if (false? (:format-rows? middleware))
```

Comment:
I think it's more common to use falsey values rather than checking for the false boolean, i.e. use `(if (:format-rows?...)` and flip your args, or `(if-not (:format-rows?`. Any reason not to do that here?

Reply:
It's so that nil (ie. value not present which is the case for most of the code base) doesn't get trigger it. But it's rather opaque. Probably better to change to use a default value.